### PR TITLE
Fixing for Genie fusion issue 5613

### LINKED ISSUE #5613: Mutated Genes table does not calculate frequencies correctly
I see this issue in the AACR GENIE study when selecting samples that have mutations in multiple different genes. I see frequencies greater than 100% sometimes.

I think this happens when selecting genes that are only covered in some samples.

image

### PR DESCRIPTION
**Pull request by khzhu**
# What? Why?
Fix #5613 

Changes proposed in this pull request:
1. backend:
 
 - [x] add a new myBatis query getSampleCountInMultipleMolecularProfilesForMutationOrFusion to retrieve Mutations or Fusions from the mutation table.
  - [x] add a new web service getSampleCountInMultipleMolecularProfilesForFusion
  - [x] modify the existing web service getSampleCountInMultipleMolecularProfiles to use the new myBastis query created.
  - [x] add a new API endpoint /fusion-genes/fetch
2. frontend:
  - [x] add a new Fusion table without Frequency column in the study view

# Checks
- [ ] Runs on Heroku.
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Google Style Guide](https://github.com/google/styleguide).
- [ ] If this is a feature, the PR is to rc. If this is a bug fix, the PR is to master.

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

### CODE REVIEW

**Comment by jjgao** [comment]
@khzhu wondering why there are 103 commits? would you rebase/squash it?

**Comment by khzhu** [comment]
@JJ, I did a rebase. Will do a squash then. thanks!

**Comment by jjgao** [comment]
@khzhu thanks!

**Comment by khzhu** [comment]
@zhx828 , I've added a new Gene filter for fusions. Please have your final review and let me know if anything else need to be done. Thanks!

File: business/src/main/resources/org/mskcc/cbio/portal/persistence/StudyMapperLegacy.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
         DESCRIPTION as description,
         PMID as pmid,
         CITATION as citation,
-        GROUPS as groups,
+        "GROUPS" as "groups",
```
Comment (n1zea144) [inline business/src/main/resources/org/mskcc/cbio/portal/persistence/StudyMapperLegacy.xml]: Why is this put here and if required, should it be added to getAllStudies in the same file?

File: persistence/persistence-api/src/main/java/org/cbioportal/persistence/MutationRepository.java [hunk 2]
```diff
@@ -23,6 +23,11 @@ MutationMeta getMetaMutationsInMolecularProfileBySampleListId(String molecularPr
                                                            Integer pageSize, Integer pageNumber,
                                                            String sortBy, String direction);
 
+    List<Mutation> getMutationsInMultipleMolecularProfilesByMutationType(List<String> molecularProfileIds, List<String> sampleIds,
+                                                                         List<Integer> entrezGeneIds, String mutationType, Boolean complementSet, String projection,
```
Comment (n1zea144) [inline persistence/persistence-api/src/main/java/org/cbioportal/persistence/MutationRepository.java]: What are the possible values of mutation type and why not use an enum?

File: business/src/main/resources/org/mskcc/cbio/portal/persistence/StudyMapperLegacy.xml (hunk 1 above)
Comment (khzhu) [reply business/src/main/resources/org/mskcc/cbio/portal/persistence/StudyMapperLegacy.xml]: We should always quote those mySql reserved key words. I agree with you any fixing should not be in this PR. Will undo the change.

File: persistence/persistence-api/src/main/java/org/cbioportal/persistence/MutationRepository.java (hunk 2 above)
Comment (khzhu) [reply persistence/persistence-api/src/main/java/org/cbioportal/persistence/MutationRepository.java]: good catch, @n1zea144! I do not think this method ever be used. Will do some testing tomorrow before removing it. thanks!

Comment (khzhu) [reply persistence/persistence-api/src/main/java/org/cbioportal/persistence/MutationRepository.java]: Hi @n1zea144 , I've done my corrections and committed the changes. Please have your final review. Thank you!

File: web/src/main/java/org/cbioportal/web/StudyViewController.java [hunk 3]
```diff
@@ -223,6 +223,43 @@
...
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface
+        @RequestAttribute(required = false, value = "involvedCancerStudies") Collection<String> involvedCancerStudies,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface. this attribute is needed for the @PreAuthorize tag above.
+        @Valid @RequestAttribute(required = false, value = "interceptedStudyViewFilter") StudyViewFilter interceptedStudyViewFilter) throws StudyNotFoundException {
+        List<SampleIdentifier> filteredSampleIdentifiers = studyViewFilterApplier.apply(interceptedStudyViewFilter);
+        List<MutationCountByGene> result = new ArrayList<>();
+        if (!filteredSampleIdentifiers.isEmpty()) {
+            List<String> studyIds = new ArrayList<>();
+            List<String> sampleIds = new ArrayList<>();
+            studyViewFilterUtil.extractStudyAndSampleIds(filteredSampleIdentifiers, studyIds, sampleIds);
```
Comment (sheridancbio) [inline web/src/main/java/org/cbioportal/web/StudyViewController.java]: The logic below seems to go beyond simple argument checking and service selection. I think this qualifies as business logic which should be encapsulated and tested in the service layer. I see that a util class is employed here ... but even still, I think this qualifies as business logic (data transformations). I would not insist on relocation, but I guess I strongly recommend keeping data transformations out of the controller handler.

File: web/src/main/java/org/cbioportal/web/parameter/StudyViewFilter.java [hunk 4]
```diff
@@ -19,11 +19,12 @@
     private List<String> studyIds;
     private List<ClinicalDataEqualityFilter> clinicalDataEqualityFilters;
     private List<ClinicalDataIntervalFilter> clinicalDataIntervalFilters;
-    private List<MutationGeneFilter> mutatedGenes;
-    private List<CopyNumberGeneFilter> cnaGenes;
-    private Boolean withMutationData;
-    private Boolean withCNAData;
-    private RectangleBounds mutationCountVsCNASelection;
+	private List<MutationGeneFilter> mutatedGenes;
+    private List<FusionGeneFilter> fusionGenes;
```
Comment (sheridancbio) [inline web/src/main/java/org/cbioportal/web/parameter/StudyViewFilter.java]: standardize indentation to uniform 4-space indents (no tabs)

File: web/src/main/java/org/cbioportal/web/parameter/StudyViewFilter.java [hunk 5]
```diff
@@ -81,8 +82,14 @@ public void setClinicalDataIntervalFilters(List<ClinicalDataIntervalFilter> clin
         return mutatedGenes;
     }
 
-    public void setMutatedGenes(List<MutationGeneFilter> mutatedGenes) {
-        this.mutatedGenes = mutatedGenes;
+	public void setMutatedGenes(List<MutationGeneFilter> mutatedGenes) { this.mutatedGenes = mutatedGenes; }
```
Comment (sheridancbio) [inline web/src/main/java/org/cbioportal/web/parameter/StudyViewFilter.java]: standardize indentation to 4-space indentation (no tabs)

**Comment by sheridancbio** [comment]
The connections to the involvedCancerStudyExtractorInterceptor class look good.

File: web/src/main/java/org/cbioportal/web/StudyViewController.java [hunk 6]
```diff
@@ -223,6 +223,43 @@
...
     }
 
+    @PreAuthorize("hasPermission(#involvedCancerStudies, 'Collection<CancerStudyId>', 'read')")
+    @RequestMapping(value = "/fusion-genes/fetch", method = RequestMethod.POST,
+        consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Fetch fusion genes by study view filter")
+    public ResponseEntity<List<MutationCountByGene>> fetchFusionGenes(
+        @ApiParam(required = true, value = "Study view filter")
+        @Valid @RequestBody(required = false) StudyViewFilter studyViewFilter,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface
```
[the remark this answers is not in the document]
Comment (khzhu) [reply web/src/main/java/org/cbioportal/web/StudyViewController.java]: @sheridancbio , thank you for the review. I will make corrections.

[the remark this answers is not in the document]
Comment (khzhu) [reply web/src/main/java/org/cbioportal/web/StudyViewController.java]: second @ApiIgnore removed.

[the remark this answers is not in the document]
Comment (khzhu) [reply web/src/main/java/org/cbioportal/web/StudyViewController.java]: corrected @ApiIgnore comments.

File: web/src/main/java/org/cbioportal/web/StudyViewController.java (hunk 3 above)
Comment (khzhu) [reply web/src/main/java/org/cbioportal/web/StudyViewController.java]: total agree and very good points! It is not just here but the whole class should be refactored to pull those data transformations out and put into respective service layer. How about we have a follow up PR instead do correction here. What do you think @sheridancbio, @zhx828?

Comment (khzhu) [reply web/src/main/java/org/cbioportal/web/StudyViewController.java]: will keep this principle in mind when switching to the new SV table and reimplementing fetch-fusion-gene endpoint.

File: web/src/main/java/org/cbioportal/web/parameter/StudyViewFilter.java (hunk 4 above)
Comment (khzhu) [reply web/src/main/java/org/cbioportal/web/parameter/StudyViewFilter.java]: corrected indents.

**Pull request merged**